tests/unit/array_property_test.js

```diff
@@ -214,39 +215,41 @@ module('unit - `MF.array` property', function (hooks) {
     );
   });
 
-  test('supports array observers', async function (assert) {
-    store.push({
-      data: {
-        type: 'person',
-        id: 1,
-        attributes: {
-          nickName: 'Tyrion Lannister',
-          titles: ['Hand of the King'],
+  if (!gte('ember-data', '4.0.0')) {
```

Comment:
Technically, this should check the ember-source version. We could also remove the code from StatefulArray which handles array observers.

Reply:
Fixed in 8f98d19ff6791e32f72eee93e737347c0b3338b2